### ISSUE TITLE
[ENH] Reorganise the structure of the clustering module

### ISSUE THREAD

**Issue by TonyBagnall** [aeon-toolkit/aeon#2252]
### Describe the feature or idea you want to propose

currently contains the following submodules

feature_based
deep_learning
averaging
compose

### Describe your proposed solution

averaging is to going to move to protoyping

after meeting on 25/10/24 propose adding

1. hierarchical (when we get a hierarchical algorithm)
2. partional for all of the k-means and medoids algorithms

**Comment by CodeLionX** [comment]
Is there a roadmap for adding hierarchical clustering algorithms?

I have some code lying around that uses [`sklearn.cluster.AgglomerativeClustering`](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.AgglomerativeClustering.html) with `metric="precomputed"` and the aeon distances. It was rather straight-forward until I wanted to use ward-linkage with any elastic distance because sklearn forbids using ward-linkage with anything other than Euclidean distance. I know it is not supposed to be used like that but provides good results.

**Comment by TonyBagnall** [comment]
No roadmap, I think @chrisholder has some kicking around t go in soon.

**Comment by TonyBagnall** [comment]
this has been done
